lib/client.js

```diff
@@ -901,6 +905,16 @@ function onSocketConnect () {
 }
 
 function onSocketError (err) {
+  const { [kParser]: parser } = this
+  if (err.code === 'ECONNRESET' && parser.statusCode && !parser.shouldKeepAlive) {
```

Comment:
```suggestion
  if (err.code === 'ECONNRESET' && parser.statusCode >= 400 && !parser.shouldKeepAlive) {
```

Reply:
Not sure about the `400`. Adding here as topic for discussion.

Reply:
why would the status code make any difference?